core-foundation/src/url.rs

```diff
@@ -77,6 +82,18 @@ impl CFURL {
         }
     }
 
+    #[cfg(unix)]
+    pub fn to_path(&self) -> PathBuf {
+        // implementing this on Windows is more complicated because of the different OsStr representation
+        unsafe {
+            let mut buf: [u8; PATH_MAX as usize] = mem::uninitialized();
+            CFURLGetFileSystemRepresentation(self.0, true as Boolean, buf.as_mut_ptr(), buf.len() as CFIndex);
```

Comment:
This returns false if an error occurred, so we should make this method return None if that occurs.